src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +73,50 @@
     }
   }
 
+  /**
+   * Get the last traded price for each Symbol that is passed in.
+   *
+   * @param symbol the symbol to get a historical prices for.
+   * @param range the range to get historical prices for
+   * @param date the date to get historical prices for
+   * @return a list of historical prices for the Symbol that is passed in.
+   */
+  public List<IexHistoricalPrices> getHistoricalPricesForSymbol(
+      final String symbol,
+      final String range,
+      final String date) {
+    if (symbol.compareTo("") == 0) {
+      return Collections.emptyList();
+    }
+
+    //handle empty parameters -- make range and date optional -- repeating myself here! Final vars
+
+    if (range == null) {
+      if (date == null) {
+        //handle filling in symbol -- for each
+        returnedHistoricalPrices = iexClientHistoricalPrices
+            .getHistoricalPricesForSymbol(symbol, "1m", "", apiKey);
+        returnedHistoricalPrices.forEach(s -> s.setSymbol(symbol));
+      } else {
+        //handle filling in symbol -- for each
+        returnedHistoricalPrices = iexClientHistoricalPrices
+            .getHistoricalPricesForSymbol(symbol, "1m", date, apiKey);
+        returnedHistoricalPrices.forEach(s -> s.setSymbol(symbol));
```

Comment:
Why is this set symbol required? Let's get to the bottom of why this isn't working correctly rather than patching it with a foreach loop that mutates a list (bad practice also)